src/inputfile.rs

```diff
@@ -59,7 +59,10 @@ impl<'a> InputFile<'a> {
         match self {
             InputFile::StdIn => Ok(InputFileReader::new(stdin.lock())),
             InputFile::Ordinary(filename) => {
-                let file = File::open(filename)?;
+                let file = match File::open(filename) {
```

Comment:
I think we could use `.map_err` here. Something like:
```rust
let file = File::open(filename).map_err(format!(...).into())?;
```

Reply:
Thanks for your advice!
I tried it, but I couldn't get it built...
Seems out of my hands...

```
62 |                 let file = File::open(filename).map_err(format!("{}: {}", filename, e).into())?;                                                    
   |                                                                                     ^ not found in this scope   
```

```
62 |                 let file = File::open(filename).map_err(|e| format!("{}: {}", filename, e).into())?;
   |                                                 ^^^^^^^ cannot infer type for `F` 
```

```
62 |                 let file = File::open(filename).map_err(|e:Error| format!("{}: {}", filename, e).into())?;  
   |                                                 ^^^^^^^ ------------------- found signature of `fn(errors::Error) -> _`
   |                                                 |                                                           
   |                                                 expected signature of `fn(std::io::Error) -> _`  
```

```
62 |                 let file = File::open(filename).map_err(|e: std::io::Error| -> std::io::Error {format!("{}: {}", filename, e).into()})?;            
   |                                                                                                                               ^^^^ the trait `std::convert::From<std::string::String>` is not implemented for `std::io::Error`
```

Reply:
Ah, I was wrong. `.into()` is not needed. This should work (I'd like to add the additional quotes around the filename):
``` rust
                let file = File::open(filename).map_err(|e| format!("'{}': {}", filename, e))?;
```